mypy/waiter.py

```diff
@@ -37,13 +37,11 @@ def __init__(self, name: str, args: List[str], *, cwd: str = None,
         self.passthrough = passthrough
 
     def start(self) -> None:
-        if self.passthrough is None:
+        if self.passthrough is None or self.passthrough < 0:
             self.outfile = tempfile.TemporaryFile()
         else:
             if self.passthrough >= 0:
```

Comment:
You don't need this `if` -- it makes it look as if there's a case where self.outfile won't be set at all.